goal_src/jak3/engine/collide/collide-cache.gc

```diff
@@ -7,3 +7,898 @@
+                                        )
+          )
+    (set! (-> arg0 exit-planes 1 y) (if (< (-> arg0 move-dist y) 0.0)
+                                        0.0
+                                        100000000000000000000000000000000000000.0
+                                        )
+          )
+    (set! (-> arg0 exit-planes 1 z) (if (< (-> arg0 move-dist z) 0.0)
+                                        0.0
+                                        100000000000000000000000000000000000000.0
+                                        )
+          )
+    (when (logtest? (-> arg0 collide-with) (collide-spec backgnd))
+      (fill-from-bg
+        this
+        (the-as
+          (function collide-hash int collide-list collide-query int)
+          (method-of-type collide-hash drawable-method-12)
+          )
+        collide-list-fill-bg-using-line-sphere
+        arg0
+        )
+      (dotimes (s4-0 (-> *level* length))
+        (let ((v1-36 (-> *level* level s4-0)))
+          (when (= (-> v1-36 status) 'active)
+            (let ((a0-18 (-> v1-36 bsp hfrag-drawable)))
+              (if (nonzero? a0-18)
+                  (hfragment-method-18 (the-as hfragment a0-18) this arg0)
+                  )
+              )
+            )
+          )
+        )
+      (let ((a0-19 (-> this num-tris)))
+        (when (> a0-19 0)
+          (let ((v1-43 (-> this prims))
+                (a1-11 *collide-shape-prim-backgnd*)
+                )
+            (set! (-> v1-43 0 num-tris) (the-as uint a0-19))
+            (set! (-> v1-43 0 prim) a1-11)
+            (set! (-> this num-prims) 1)
+            (set! (-> v1-43 0 first-tri) (the-as uint 0))
+            (set! (-> v1-43 0 ccache) this)
+            (set! (-> v1-43 0 prim-core world-sphere quad) (-> a1-11 prim-core world-sphere quad))
+            (set! (-> v1-43 0 prim-core quad 1) (-> a1-11 prim-core quad 1))
+            )
+          )
+        )
+      (+! (-> *collide-stats* output) (-> this num-tris))
+      )
+    (when (logtest? (-> arg0 collide-with) (collide-spec water))
+      (let ((a1-13 (-> (the-as process-drawable (-> arg0 ignore-process0)) water)))
+        (if (nonzero? a1-13)
+            (fill-from-water this a1-13)
+            )
+        )
+      )
+    (if (logtest? (-> arg0 collide-with) (collide-spec hit-by-player-list hit-by-others-list player-list))
+        (fill-from-fg-line-sphere this arg0)
+        )
+    0
+    (label cfg-55)
+    (none)
+    )
+  )
+
+(defmethod fill-from-fg-boxes ((this collide-cache))
+  (let ((s5-0 (-> this collide-with)))
+    (set! *actor-list-length* 0)
+    (when (logtest? s5-0 (collide-spec hit-by-others-list))
+      (let* ((a0-1 *actor-hash*)
+             (t9-0 (method-of-object a0-1 add-an-object))
+             (a1-0 (-> this collide-box))
+             (a2-0 *actor-list*)
+             )
+        256
+        (set! *actor-list-length* (t9-0 a0-1 (the-as vector a1-0) (the-as hash-object-info a2-0)))
```

Comment:
I think we have the methods wrong on actor-hash here - the 256 should be an arg, and the casts seem wrong. fine to leave it if we haven't decomped actor-hash yet.